audio_recognize.py

```diff
@@ -57,37 +57,29 @@ def cluster_syllables(syllables, specgram, sp_freq,
     max_dur = 0
     test = []
     syllables_final = []
-    # high_thres = 0.015
-    # low_thres = 0.006
-    # print(len(specgram))
-    # print(len(syllables))
+
     if not train:
         kmeans_centers = np.load('kmeans_centers.npy')
+    vec1 = []
     for syl in syllables:
         # for each detected syllable (vocalization)
 
         # A. get the spectrogram area in the defined frequency range
         start = int(syl[0] / win)
         end = int(syl[1] / win)
         
-        # np.save('segments.npy', segments)
         cur_image = specgram[start:end, f1:f2]
         if cur_image.shape[0]==0 or cur_image.shape[1]==0:
             continue
         temp_image = cur_image/np.amax(cur_image)
         if train:
             images.append(cur_image)
             continue
-        vec = [np.mean(temp_image),np.var(temp_image), np.mean(cur_image-np.amax(cur_image)), np.var(cur_image-np.amax(cur_image))]
-        if np.linalg.norm(vec-kmeans_centers[1]) < np.linalg.norm(vec-kmeans_centers[0]):
-            # print(mentemp_image)
-            # print([start, end])
-            # plt.imshow(temp_image.T)
-            # plt.show()
+        vec=[np.mean(temp_image),np.var(temp_image), np.mean(cur_image-np.amax(cur_image)), np.var(cur_image-np.amax(cur_image))]
+        if (vec[0]-0.7*kmeans_centers[1,0]>=0 and vec[1]-0.7*kmeans_centers[1,2]>=0):
```

Comment:
could you explain this rule here and why is this repeated in main_live as well? maybe we should make it a function?